src/Features/Core/Portable/NavigateTo/AbstractNavigateToSearchService.CachedDocumentSearch.cs

```diff
@@ -120,84 +118,56 @@ private static void ClearCachedData()
 
         using var _1 = GetPooledHashSet(priorityDocumentKeys, out var priorityDocumentKeysSet);
 
-        // Sort the groups into a high pri group (projects that contain a high-pri doc), and low pri groups (those
-        // that don't).
+        // Sort the groups into a high pri group (projects that contain a high-pri doc), and low pri groups (those that
+        // don't), and process in that order.
         using var _2 = GetPooledHashSet(groups.Where(g => g.Any(priorityDocumentKeysSet.Contains)), out var highPriorityGroups);
         using var _3 = GetPooledHashSet(groups.Where(g => !highPriorityGroups.Contains(g)), out var lowPriorityGroups);
```

Comment:
Don't understand why using hashsets here, why not just:

var highPriorityGroups = groups.Where(g => g.Any(priorityDocumentKeysSet.Contains));
var lowPriorityGroups = groups.Where(g => !highPriorityGroups.Contains(g))

Reply:
I'll change. It's just how things were done before.

Reply:
ok.  i remember why.  the reason here is that the first check already walked everything (including *into* the groups).  the second line doesn't need to walk into the groups at all.  The low priority groups are determined by just checking them against the high pri groups.  using sets makes the second pass O(1) for each group.